pkg/cortex/modules.go

```diff
@@ -514,29 +514,28 @@ func (t *Cortex) setupModuleManager() error {
 
 	// Register all modules here.
 	// RegisterModule(name string, initFn func()(services.Service, error))
-	mm.RegisterModule(Server, t.initServer)
-	mm.RegisterModule(API, t.initAPI)
-	mm.RegisterModule(RuntimeConfig, t.initRuntimeConfig)
-	mm.RegisterModule(MemberlistKV, t.initMemberlistKV)
-	mm.RegisterModule(Ring, t.initRing)
-	mm.RegisterModule(Overrides, t.initOverrides)
-	mm.RegisterModule(Distributor, t.initDistributor)
-	mm.RegisterModule(Store, t.initStore)
-	mm.RegisterModule(DeleteRequestsStore, t.initDeleteRequestsStore)
-	mm.RegisterModule(Ingester, t.initIngester)
+	mm.RegisterModule(Server, t.initServer, modules.PrivateModule)
+	mm.RegisterModule(API, t.initAPI, modules.PrivateModule)
+	mm.RegisterModule(RuntimeConfig, t.initRuntimeConfig, modules.PrivateModule)
+	mm.RegisterModule(MemberlistKV, t.initMemberlistKV, modules.PrivateModule)
+	mm.RegisterModule(Ring, t.initRing, modules.PrivateModule)
+	mm.RegisterModule(Overrides, t.initOverrides, modules.PrivateModule)
+	mm.RegisterModule(Store, t.initStore, modules.PrivateModule)
+	mm.RegisterModule(DeleteRequestsStore, t.initDeleteRequestsStore, modules.PrivateModule)
+	mm.RegisterModule(StoreQueryable, t.initStoreQueryable, modules.PrivateModule)
+	mm.RegisterModule(StoreGateway, t.initStoreGateway)
+	mm.RegisterModule(Purger, t.initPurger)
+	mm.RegisterModule(Compactor, t.initCompactor)
 	mm.RegisterModule(Flusher, t.initFlusher)
-	mm.RegisterModule(Querier, t.initQuerier)
-	mm.RegisterModule(StoreQueryable, t.initStoreQueryable)
 	mm.RegisterModule(QueryFrontend, t.initQueryFrontend)
 	mm.RegisterModule(TableManager, t.initTableManager)
 	mm.RegisterModule(Ruler, t.initRuler)
 	mm.RegisterModule(Configs, t.initConfig)
 	mm.RegisterModule(AlertManager, t.initAlertManager)
-	mm.RegisterModule(Compactor, t.initCompactor)
-	mm.RegisterModule(StoreGateway, t.initStoreGateway)
-	mm.RegisterModule(Purger, t.initPurger)
+	mm.RegisterModule(Querier, t.initQuerier)
+	mm.RegisterModule(Ingester, t.initIngester)
+	mm.RegisterModule(Distributor, t.initDistributor)
```

Comment:
It would be nice to minimize diff here to make changed parts visible. (Thanks for fixing double-registration of StoreGateway!)